dwave/optimization/src/nodes/manipulation.cpp

```diff
@@ -18,6 +18,154 @@
 
 namespace dwave::optimization {
 
+std::vector<ssize_t> make_concatenate_shape(std::vector<ArrayNode*> array_ptrs, ssize_t axis) {
+    // One or more arrays must be given
+    if (array_ptrs.size() < 1) {
+        throw std::invalid_argument("need at least one array to concatenate");
+    }
+
+    for (auto it = std::next(array_ptrs.begin()), stop = array_ptrs.end(); it != stop; it++) {
```

Comment:
```c++
    for (auto it = std::next(array_ptrs.begin()), stop = array_ptrs.end(); it != stop; ++it) {
```
almost certainly doesn't matter in this case, but [preincrement is a good habit](https://llvm.org/docs/CodingStandards.html#prefer-preincrement). Here and elsewhere.

Reply:
fixed